lib/phoenix/controller.ex

```diff
@@ -206,16 +217,29 @@ defmodule Phoenix.Controller do
 
   @doc """
   Stores the view for rendering.
+
+  Raises `Phoenix.ControllerAlreadySentError` if the conn was already sent.
   """
   @spec put_view(Plug.Conn.t, atom) :: Plug.Conn.t
+  def put_view(%Plug.Conn{state: state}, _module) when not state in @unsent do
```

Comment:
Don't change to `@sent` because we may introduce new sent states in Plug. Furthermore, `:file` is also considered as sent.

Checking for `@unsent` as today is the best way to go. We can remove the double negative by reordering the clauses.